lib/puma/rack_default.rb

```diff
@@ -2,8 +2,20 @@
 
 require_relative '../rack/handler/puma'
 
-module Rack::Handler
-  def self.default(options = {})
-    Rack::Handler::Puma
+if Object.const_defined? :Rackup
```

Comment:
Although this works, I think testing for Rack version 3 is more meaningful, because this tells the reader that this was implemented in response to a change in the new Rack version.

Reply:
Sorry, I thought I responded.  The rackup namespace needs to be defined.  The Rack v3 gem could be loaded, but unless the rackup gem is also loaded, this will fail.  I added a comment.